### PR TITLE
[REVIEW] Add `copyright` check in `cudf`

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #https://github.com/rapidsai/cudf/issues/9729

This PR adds copyright check to cudf, this is taken from https://github.com/rapidsai/cuml

### CODE REVIEW

**Review by raydouglass — APPROVED** [review]

File: python/cudf/setup.py
Comment (vyasr) [inline python/cudf/setup.py:2]: But eventually unmodify on purpose please :)

Comment (galipremsagar) [reply python/cudf/setup.py:2]: yeah looking into it..

Comment (ttnghia) [reply python/cudf/setup.py:2]: It seems that this needs to be merged first before it can run?

Comment (vyasr) [reply python/cudf/setup.py:2]: From a quick look through the logic in the copyright I'm not sure that it will flag this as an error. I think you have to either 1) have an invalid year, 2) have a later year before an earlier year (e.g. 2020-2018), or 3) if there is a line before the copyright line. I don't think it is a general check for the entire copyright.

**Review by vyasr — APPROVED** [review]
Thanks for adding this! I think you'll need to make a different change to test that the check is working though.

A couple of notes that I don't intend to hold up this PR for:
- I'm not a huge fan of the fact that we just copy a lot of CI scripts between repos and it would be nice to have a way to share this code more effectively (maybe an ops-maintained submodule containing helpers), but I'm OK pushing that off to get this check enabled here for now. 
- We really should be using [GitPython](https://gitpython.readthedocs.io/en/stable/) instead of writing our own gitutils.

File: python/cudf/setup.py
Comment (vyasr) [reply python/cudf/setup.py:2]: See my review. I don't think this is the kind of error that this script will catch.

Comment (galipremsagar) [reply python/cudf/setup.py:2]: It does catch locally, but not on CI. So looking into what is happening here..

File: python/cudf/setup.py
Comment (ttnghia) [inline python/cudf/setup.py:1]: No no please keep 2021. It should not pass for any year that is not 2022.

Comment (galipremsagar) [reply python/cudf/setup.py:1]: I'm just testing in CI, it will not pass. Will change to prove for 2021 as well next

Comment (ttnghia) [reply python/cudf/setup.py:1]: Kudo for 2020:
```
FAILED: copyright check; end output
```

Comment (galipremsagar) [reply python/cudf/setup.py:1]: Failed for 2020: https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-style/23578/console

Comment (galipremsagar) [reply python/cudf/setup.py]: Failed for 2021 too: https://gpuci.gpuopenanalytics.com/blue/organizations/jenkins/rapidsai%2Fgpuci%2Fcudf%2Fprb%2Fcudf-style/detail/cudf-style/23579/pipeline

File: python/cudf/setup.py
Comment (galipremsagar) [reply python/cudf/setup.py]: Removed this on purpose :P

**Review by ttnghia — APPROVED** [review]

**Comment by galipremsagar** [comment]
> Thanks for adding this! I think you'll need to make a different change to test that the check is working though.
> 
> A couple of notes that I don't intend to hold up this PR for:
> 
> * I'm not a huge fan of the fact that we just copy a lot of CI scripts between repos and it would be nice to have a way to share this code more effectively (maybe an ops-maintained submodule containing helpers), but I'm OK pushing that off to get this check enabled here for now.
> * We really should be using [GitPython](https://gitpython.readthedocs.io/en/stable/) instead of writing our own gitutils.

cc: @rapidsai/ops-codeowners something we could centralize in a common repo(the .py code) with minimal config variables that each repo can have control over. Not part of this PR but something to note for the future.

**Comment by vyasr** [comment]
> cc: @rapidsai/ops-codeowners something we could centralize in a common repo(the .py code) with minimal config variables that each repo can have control over. Not part of this PR but something to note for the future.

That also applies to lots of other files in the `ci` directory of our repos. Just an idea, not urgent to implement.

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**
